dotrix_core/src/assets/mesh.rs

```diff
@@ -44,11 +44,38 @@ impl Mesh {
         self.layout.push(format);
     }
 
-    /// Sets indicies to the mesh
+    /// Get vertices with type casting
+    pub fn vertices_as<T>(&self, index: usize) -> Vec<T>
```

Comment:
both methods should return Iterators instead.

Reply:
should I simply remove .collect or you see it somehow more complicated?

Reply:
I would make a wrapper around the vector's mapped iterator